packages/fx-core/src/common/spec-parser/utils.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+"use strict";
+
+import axios from "axios";
+import fs from "fs-extra";
+import { ConstantString } from "./constants";
+import { OpenAPIV3 } from "openapi-types";
+
+export async function isYamlSpecFile(specPath: string): Promise<boolean> {
+  const isRemoteFile = specPath.startsWith("http:") || specPath.startsWith("https:");
```

Comment:
Can we just depend on the extension name to identify yml or json?

Reply:
For remote url, maybe there is no extension name. So just use extension name is not enough, it is better to parse it for accurate result.

Reply:
I can optimize this function, only parse the file when the extension name cannot determine the format.